src/hash.rs

```diff
@@ -84,16 +84,42 @@ pub struct HashInfo {
     /// The hash (first 7 letters of the sha1sum)
     pub hash: String,
 
-    /// The exact input that was used to compute the hash
-    pub hash_input: String,
-
     /// The hash prefix (e.g. `py38` or `np111`)
-    pub hash_prefix: String,
+    #[serde(default, skip_serializing_if = "String::is_empty")]
+    pub prefix: String,
+}
+
+/// Represents the input to compute the hash
+pub struct HashInput(String);
+
+impl HashInput {
+    /// Create a new hash input from a variant
+    pub fn from_variant(variant: &BTreeMap<String, String>) -> Self {
+        let mut buf = Vec::new();
```

Comment:
do you think it's worth to make vec with capacity of BtreeMAP keys?

Reply:
I doubt that will increase performance significantly, and its also not a critical path. Id opt for simplicity instead.